### PR TITLE
PYTHON-4429 Add Changelog for 4.7.2

### PR DESCRIPTION
**Pull request by blink1073**
(cherry picked from commit 29f4d5cf8952b1abe0a73b11631ff84818b7f9ed)

### CODE REVIEW

File: doc/changelog.rst [hunk 1]
```diff
@@ -6,6 +6,22 @@ Changes in Version 4.8.0
 
 .. warning:: PyMongo 4.8 drops support for Python 3.7 and PyPy 3.8: Python 3.8+ or PyPy 3.9+ is now required.
 
+Changes in Version 4.7.2
+-------------------------
+
+Version 4.7.2 fixes a bug introduced in 4.7.0:
```
Comment (caseyclements) [inline doc/changelog.rst:12]: This is ambiguous. Is there just the Nuitka compiler bug?

File: doc/changelog.rst [hunk 2]
```diff
@@ -6,6 +6,22 @@ Changes in Version 4.8.0
...
+
+- Fixed a bug where PyMongo could not be used with the Nuitka compiler.
+
+Issues Resolved
+...............
+
+See the `PyMongo 4.7.2 release notes in JIRA`_ for the list of resolved issues
```
Comment (caseyclements) [inline doc/changelog.rst:19]: There's something funny with these links. For one, they not automatically take you to them by clicking in github. But more importantly, when I paste the link into my browser, I don't see things rendered as html. I see raw tags, like <h2>.

**Review by caseyclements — CHANGES_REQUESTED** [review]

File: doc/changelog.rst (hunk 2 above)
Comment (blink1073) [reply doc/changelog.rst:19]: We're moving away from this changelog toward the "What's New" page on the internal docs site.

File: doc/changelog.rst (hunk 1 above)
Comment (blink1073) [reply doc/changelog.rst:12]: Yes

Comment (caseyclements) [reply doc/changelog.rst:12]: How about just one line:

Fixes a bug where PyMongo could not be used with the Nuitka compiler.

File: doc/changelog.rst (hunk 2 above)
Comment (caseyclements) [reply doc/changelog.rst:19]: So this is how it's supposed to look?

Comment (blink1073) [reply doc/changelog.rst:19]: 🤷🏼‍♂️

File: doc/changelog.rst (hunk 1 above)
Comment (blink1073) [reply doc/changelog.rst:12]: I want to convey that the bug was introduced in 4.7.0.  It worked in prior versions.

**Review by caseyclements — APPROVED** [review]
Git-R-Done!

**Pull request merged**